File: irrgarten/src/irrgarten/GameState.java
package irrgarten;

/**
 * Clase GameState. Implementación de la clase GameState con sus métodos y variables.
 * 
 * @author Miguel Ángel Luque Gómez
 * correo e.mangelluqg@go.ugr.es
 * 
 * @author Sergio Calvo González
 * correo e.sergiocg10@go.ugr.es
 * 
 */
public class GameState {
    private String labyrinth;   //Laberinto del juego
    private String players;     //Jugadores del juego
    private String monsters;    //Monstruos del juego
    private int currentPlayer;  //Índice del jugador actual
    private boolean winner;     //Variable booleana que muestra si hay o no ganador
    private String log;         //Variable log, donde guardar eventos anteriores
    
    /**
     * Constructor por parámetro de la clase GameState.
     * @param lab   representación del laberinto
     * @param pl    representación de los jugadores
     * @param monst representación de los monstruos
     * @param cp    índice del jugador actual
     * @param w     variable que muestra si hay o no ganador
     * @param l     variable donde guardar eventos del anterior turno
     */
    public GameState(String lab, String pl, String monst, int cp, boolean w, String l){
        labyrinth=lab;
        players=pl;
        monsters=monst;
        currentPlayer=cp;
        winner=w;
        log=l;
    }
    
    /**
     * Funcion getLabyrinth. Función getter de la variable labyrinth de la clase
     * @return Representación del laberinto
     */
    public String getLabyrinth(){
        return labyrinth;
    }
    
    /**
     * Funcion getPlayers. Función getter de la variable players de la clase
     * @return Representación de los jugadores
     */
    public String getPlayers(){
        return players;
    }
    
    /**
     * Funcion getMonsters. Función getter de la variable monsters de la clase
     * @return Representación de los monstruos
     */
    public String getMonsters(){
        return monsters;
    }
    
    /**
     * Funcion getCurrentPlayer. Función getter de la variable currentPlayer de la clase
     * @return índice del jugador actual
     */
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    
    /**
     * Funcion getWinner. Función getter de la variable winner de la clase
     * @return Variable winner, que representa si hay ganador (true) o si no (false)
     */
    public boolean getWinner(){
        return winner;
    }
    
    /**
     * Funcion getLog. Función getter de la variable log de la clase
     * @return Variable donde almacenamos los eventos anteriores
     */
    public String getLog(){
        return log;
    }
}
    

File: Irrgarten/lib/GameState.rb
# Clase Player

# Clase Player. Representación de la clase player con sus métodos y variables 
# 
# @author Miguel Ángel Luque Gómez
# correo e.mangelluqg@go.ugr.es
  
# @author Sergio Calvo González
# correo e.sergiocg10@go.ugr.es
 
class GameState
		
	#Constructor de la clase GameState
	def initialize(lab, pl, monst, cp, w, l)
		@labyrinth=lab
		@players=pl
		@monsters=monst
		@currentPlayer=cp
		@winner=w
		@log=l
	end
	#Método obtención labyrinth clase GameState
	def getLabyrinth
		return @labyrinth
	end
	#Método obtención players clase GameState
	def getPlayers
		return @players
	end
	#Método obtención monsters clase GameState
	def getMonsters
		return @monsters
	end
	#Método obtención currentPlayer clase GameState
	def getCurrentPlayer
		return @currentPlayer
	end
	#Método obtención winner clase GameState
	def getWinner
		return @winner
	end
	#Método obtención log clase GameState
	def getLog
		return @log
	end
end

